docs/docs/ProgrammingGuide/Model/Serialization.MD

```diff
@@ -0,0 +1,131 @@
+                            ModuleSerializable
+                                     .
+                                     .
+            ........................................................................
+            .               .              .                       .               . 
+            .               .              .                       .               .
+     ModuleSerializer CellSerializer  ContainerSerializable   KerasSerializer  [UserDefinedSerializer]
+                                           .
+                                           .
+                               ..........................
+                               .                        .
+                               .                        .
+                         ContainerSerializer  [UserDefinedContainerSerializder] 
+                                        
+                                  
+                             
+        
+ 
+ * ModuleSerializable: abstract class to define serialization methods and provide a default implementation
+ * ModuleSerializer: entry for all layers' serializers
+ * CellSerializer: Default implementation for RNN cell modules
+ * ContainerSerializable: Abstract class for container serializers like Sequential and provide a default implementation
+ * ContainerSerializer: Default serializer for containers
+ * KerasSerializer: Keras adapter serializer implementation for keras compatible layers
+ 
+ Users can extend ModuleSerializable or ContainerSerializable to implement optional serializers for your own layer or containers
+ 
+### Supported data types
+
+Below is the data types supported in serialization
+
+* Int
+* Long
+* Short
+* Float
+* Double
+* Boolean
+* String
+* Regularizer
+* DataFormat
+* VariableFormat
+* Shape
+* InitializationMethod
+* Tensor
+* Module
+* Array
+* Map
+* Customized Data
+
+#### Implement customized data converter
+if you have your own defined data types that are not supported in serialization
+or cannot be indirectly supported by above types, you can also define your own data
+converter by extending trait `DataConverter`, which has two abstract methods to implement
+
+The `setAttributeValue` is to define how to set your own object value to attributeBuilder
+
+```scala
+ def setAttributeValue[T : ClassTag](context: SerializeContext[T],
+                                      attributeBuilder : AttrValue.Builder, value: Any,
+                                      valueType: universe.Type = null)
+    (implicit ev: TensorNumeric[T]) : Unit
+
+```
+
+In opposite you should implement `getAttributeValue` to get value from attibute
+
+```scala
+ def getAttributeValue[T : ClassTag](context: DeserializeContext,
+                                      attribute: AttrValue)(
+    implicit ev: TensorNumeric[T]) : AnyRef
+```
+Check [BigDL](https://github.com/intel-analytics/BigDL)`com.intel.analytics.bigdl.utils.serializer.converters.DataConverter`
```

Comment:
Shall we directly provide a link to `DataConverter` class?

Reply:
the doc will be released to different URL, so we should be pointing it to a abs path, but doc is release specific, we've no idea which release to point it to...

Reply:
Oh yes. Then just use this way :)